lib/krikri/parsers/xml_parser.rb

```diff
@@ -61,9 +61,81 @@ def values?
 
       private
 
-      def get_child_nodes(name)
-        Krikri::Parser::ValueArray.new(@node.xpath("#{@node.path}/#{name}", @namespaces)
-          .map { |node| self.class.new(node, @namespaces) })
+      ##
+      # Return a Krikri::Parser::ValueArray of child nodes that match a
+      # name or "name expression" as interpreted by #xpath_predicate
+      #
+      # @see Krikri::JsonParser#get_child_nodes
+      # @see Krikri::XmlParser::Value#xpath_expression
+      # @see Krikri::XmlParser::Value#xpath_predicate
+      #
+      # @example
+      #    va = value.get_child_nodes('dc:subject')
+      #    va = value.get_child_nodes('dc:this|dc:that')
+      #
+      # @param name_exp [String]  Element name expression
+      # @return [Krikri::Parser::ValueArray]
+      def get_child_nodes(name_exp)
+        Krikri::Parser::ValueArray.new(
+          @node.xpath(xpath_expression(name_exp), @namespaces)
+            .map { |node| self.class.new(node, @namespaces) }
+        )
+      end
+
+      ##
+      # Return a string for an XPATH expression
+      #
+      # The resulting path string will produce the element that occurs first in
+      # the XML document, which matches one of the given names.
+      #
+      # @param name_exp [String]  Element name expression
+      # @return [String]
+      def xpath_expression(name_exp)
+        #
+        # First, we have to figure out the base path.
+        #
+        # An assumption is made that no parser is really going to be used in
+        # a mapping without a root path (as defined by the root_path
+        # parameter to its constructor).  The usual XPATH expression will
+        # tend to be the first one.  The predicate will not work with
+        # @node.path == '/*', which is the default when the root path is not
+        # specified.
+        #
+        # When the root path is specified, name values with the "xmlns" prefix
+        # will not work; since this is the default namespace, we want to refer
+        # to them as the un-prefixed name.  For example, inside an OAI-PMH
```

Comment:
> Can you actually help me understand why "xmlns:" would have to be used to prefix the setSpec element name? If setSpec exists in the default namespace of the document, why can't we refer to it as setSpec?

I think this is one of those really hard to follow things about XML namespaces.  I'll try tell it how I understand it, with the caveat that I sometimes get turned around on this...

When you do `name()` you're asking for the QName of the tag __in context__ of the node's position in the document.  So a node named `http://example.org/my_tag` gets squashed to `ex:my_tag` or `my_tag`, depending on the namespace declarations that apply to the context.

An XPath expression, by contrast, is independent of the document context.  The namespaces applied are the ones declared as in scope for the _expression_.  This way a given expression finds semantically equivalent nodes across documents with different namespaces or namespace scoping.  
In XPath 1.0 the expression context is said to include ["the set of namespace declarations in scope for the expression"](http://www.w3.org/TR/xpath/#section-Introduction).  There's no default namespace here, just a convention to use "xmlns" when you know you're querying the default namespace in the document/element context.

In 2.0, namespace declarations are part of the "Static Context" of the expression (see [XPath 2.0 sec. 2.1.1](http://www.w3.org/TR/xpath20/#static_context)), and there is the possibility for a default namespace on the expression.

I'm not sure what any of this means for these changes, but I'm guessing there's a way to tighten this up a bit.  One issue that comes to mind is that using `name()` may be problematic if someone's XML has different namespace declarations/scoping across records.